File: app/src/main/java/com/iutcalendar/swiping/TouchGestureListener.java
package com.iutcalendar.swiping;

import android.content.Context;
import android.view.GestureDetector;
import android.view.MotionEvent;
import android.view.View;

public class TouchGestureListener implements View.OnTouchListener {

    private final GestureDetector gestureDetector;

    public TouchGestureListener(Context ctx, GestureEventManager gs) {
        gestureDetector = new GestureDetector(ctx, gs);
    }


    @Override
    public boolean onTouch(View v, MotionEvent event) {
        v.performClick();
        return gestureDetector.onTouchEvent(event);
    }

}

File: app/src/main/java/com/iutcalendar/swiping/TouchGestureListener.kt
package com.iutcalendar.swiping

import android.content.Context
import android.view.GestureDetector
import android.view.MotionEvent
import android.view.View
import android.view.View.OnTouchListener

class TouchGestureListener(ctx: Context?, gs: GestureEventManager?) : OnTouchListener {
    private val gestureDetector: GestureDetector

    init {
        gestureDetector = GestureDetector(ctx, gs)
    }

    override fun onTouch(v: View, event: MotionEvent): Boolean {
        v.performClick()
        return gestureDetector.onTouchEvent(event)
    }
}